spec/lib/sentry/processor/filter_request_body_spec.rb

```diff
@@ -3,47 +3,54 @@
 require 'rails_helper'
 
 RSpec.describe Sentry::Processor::FilterRequestBody do
-  let(:client) { double('client') }
-  let(:processor) { Sentry::Processor::FilterRequestBody.new(client) }
-  let(:result) { processor.process(sentry_data) }
+  context 'with PII in the [:request][:data] hash' do
+    before(:each) do
+      client = double('client')
+      @processor = Sentry::Processor::FilterRequestBody.new(client)
+    end
 
-  let(:sentry_data) do
-    {
-      'tags' => { 'controller_name' => 'ppiu', 'sign_in_method' => { 'service_name' => 'idme', 'acct_type' => nil } },
-      'request' =>
-       {
-         'data' =>
-          "{\n  \"account_type\": \"Checking\"}"
-       }
-    }
-  end
+    it 'filters PII for a controller found in FILTERED_CONTROLLER' do
+      sentry_request = create_sentry_request_with_pii(controller: 'ppiu')
+      result = @processor.process(sentry_request)
 
-  def self.expect_filter(filtered)
-    it "should#{filtered ? '' : "n't"} filter the request body" do
-      expect(result['request']['data']).public_send(
-        filtered ? 'to' : 'to_not',
-        eq(Sentry::Processor::PIISanitizer::FILTER_MASK)
-      )
+      expect(result['request']['data']).to eql(Sentry::Processor::PIISanitizer::FILTER_MASK)
     end
-  end
 
-  context 'with data from a controller in FILTERED_CONTROLLER' do
-    expect_filter(true)
-  end
+    it 'does not filter PII for a contoller not included in FILTERED_CONTROLLER' do
+      sentry_request = create_sentry_request_with_pii(controller: 'another_controller')
 
-  context 'with data from another controller' do
-    before do
-      sentry_data['tags']['controller_name'] = 'health_care_applications'
+      result = @processor.process(sentry_request)
+
+      expect(result['request']['data']).not_to eql(Sentry::Processor::PIISanitizer::FILTER_MASK)
+      expect(result['request']['data']).to eql("{\n  \"account_type\": \"Checking\"}")
     end
 
-    expect_filter(false)
+    it 'works when there is no request body' do
+      sentry_request =
+        {
+          'tags' => {
+            'controller_name' => 'ppiu',
+            'sign_in_method' => { 'service_name' => 'idme', 'acct_type' => nil }
+          },
+          'request' => {}
+        }
+      result = @processor.process(sentry_request)
+
+      expect(result['request']['data']).to eql(nil)
+    end
   end
 
-  context 'with no request body' do
-    before do
-      sentry_data['request'].delete('data')
-    end
+  private
 
-    expect_filter(false)
+  def create_sentry_request_with_pii(controller: 'default')
+    {
+      'tags' => {
+        'controller_name' => controller,
+        'sign_in_method' => { 'service_name' => 'idme', 'acct_type' => nil }
+      },
+      'request' => {
+        'data' => "{\n  \"account_type\": \"Checking\"}"
```

Comment:
Is the whitespace (`\n`) important?

Reply:
I'm not sure if this request body is strict or where it was generated/written, but ultimately I'd expect the processor to process the filtering regardless of excess `\n`